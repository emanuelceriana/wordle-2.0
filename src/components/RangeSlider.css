.rangeSlider {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 2em;
  position: relative;
}

.rangeSlider .label {
  line-height: 0;
  margin-right: 0.5em;
}

.rangeSlider .input {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 60%;
  height: 2em;
}

.rangeSlider input {
  width: 100%;
}

.rangeSlider .counter {
  position: absolute;
  top: -1em;
}
