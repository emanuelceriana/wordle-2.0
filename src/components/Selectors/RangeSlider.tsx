import { ChangeEvent, useState } from "react";
import styles from "./RangeSlider.module.scss";
import { defaultMinTriesAmount, defaultMaxTriesAmount } from "./constants";

interface RangeSliderProps {
  defaultValue: number;
  onChangeRequest: (value: number) => void;
}

export const RangeSlider = ({
  defaultValue,
  onChangeRequest,
}: RangeSliderProps) => {
  const [triesCount, setTriesCount] = useState<number>(defaultValue);

  const handleChangeRangeSlider = (e: ChangeEvent<HTMLInputElement>) => {
    const value = parseInt(e.target.value);
    if (value >= 1 && value <= 50) {
      setTriesCount(parseInt(e.target.value));
      onChangeRequest(parseInt(e.target.value));
    }
  };

  return (
    <div className={styles.rangeSlider}>
      <span className={styles["rangeSlider__label"]}>Try Count:</span>
      <div className={styles["rangeSlider__input"]}>
        <span>{defaultMinTriesAmount}</span>
        <input
          type="range"
          onChange={handleChangeRangeSlider}
          min={defaultMinTriesAmount}
          max={defaultMaxTriesAmount}
          step={1}
          value={triesCount}
        ></input>
        <span>{defaultMaxTriesAmount}</span>
        <div className={styles["rangeSlider__counter"]}>{triesCount}</div>
      </div>
    </div>
  );
};
