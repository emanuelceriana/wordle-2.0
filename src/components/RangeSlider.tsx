import { ChangeEvent, useState } from "react";
import "./RangeSlider.css";

interface RangeSliderProps {
  defaultValue: number;
  onChangeRequest: (value: number) => void;
}

export const RangeSlider = ({
  defaultValue,
  onChangeRequest,
}: RangeSliderProps) => {
  const [tryCount, setTryCount] = useState<number>(defaultValue);

  const handleChangeRangeSlider = (e: ChangeEvent<HTMLInputElement>) => {
    setTryCount(parseInt(e.target.value));
    onChangeRequest(parseInt(e.target.value));
  };

  return (
    <div className="rangeSlider">
      <span className="label">Try Count:</span>
      <div className="input">
        <span>1</span>
        <input
          type="range"
          onChange={handleChangeRangeSlider}
          min={1}
          max={50}
          step={1}
          value={tryCount}
          className="custom-slider"
        ></input>
        <span>50</span>
        <div className="counter">{tryCount}</div>
      </div>
    </div>
  );
};
