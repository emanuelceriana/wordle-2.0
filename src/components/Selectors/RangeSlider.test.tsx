import { render, fireEvent } from "@testing-library/react";
import { RangeSlider } from "./RangeSlider";
import userEvent from "@testing-library/user-event";

describe("<RangeSlider />", () => {
  const defaultValue = 6;
  const setDefaultValue = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render without errors", () => {
    const { asFragment } = render(
      <RangeSlider
        defaultValue={defaultValue}
        onChangeRequest={setDefaultValue}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render with default value", () => {
    const { getByRole } = render(
      <RangeSlider
        defaultValue={defaultValue}
        onChangeRequest={setDefaultValue}
      />
    );

    const input = getByRole("slider") as HTMLInputElement;

    expect(parseInt(input.value)).toBe(defaultValue);
  });
  it("should change value correctly", () => {
    const newValue = "20";
    const { getByRole } = render(
      <RangeSlider
        defaultValue={defaultValue}
        onChangeRequest={setDefaultValue}
      />
    );

    fireEvent.change(getByRole("slider"), { target: { value: newValue } });

    expect(setDefaultValue).toHaveBeenCalledWith(parseInt(newValue));
  });
  it("should change only between 1 and 50", () => {
    const setDefaultValue = jest.fn();
    const { getByRole } = render(
      <RangeSlider
        defaultValue={defaultValue}
        onChangeRequest={setDefaultValue}
      />
    );

    const input = getByRole("slider");
    input.setAttribute("min", "0");
    input.setAttribute("max", "1000");

    fireEvent.change(input, { target: { value: 100 } });
    expect(setDefaultValue).toHaveBeenCalledTimes(0);
    fireEvent.change(input, { target: { value: 0 } });
    expect(setDefaultValue).toHaveBeenCalledTimes(0);
  });
});
