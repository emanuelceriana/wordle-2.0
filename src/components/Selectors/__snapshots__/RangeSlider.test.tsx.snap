// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeSlider /> should render without errors 1`] = `
<DocumentFragment>
  <div
    class="rangeSlider"
  >
    <span
      class="rangeSlider__label"
    >
      Try Count:
    </span>
    <div
      class="rangeSlider__input"
    >
      <span>
        1
      </span>
      <input
        max="50"
        min="1"
        step="1"
        type="range"
        value="6"
      />
      <span>
        50
      </span>
      <div
        class="rangeSlider__counter"
      >
        6
      </div>
    </div>
  </div>
</DocumentFragment>
`;
